vmm/src/vm.rs

```diff
@@ -1616,6 +1720,41 @@ impl<'a> Vm<'a> {
         // Unblock all CPU threads.
         vcpu_thread_barrier.wait();
 
+        let mut signals: [i32; MAX_SIGNAL_VAL] = [0; MAX_SIGNAL_VAL];
+        let mut index = 0;
+        for sig_num in 1..MAX_SIGNAL_VAL {
+            if self.devices.signal_present[sig_num] {
+                signals[index] = sig_num as i32;
+                index += 1;
+            }
+        }
+        if index > 0 {
+            if let Some(console_input) = &self.devices.console_input {
+                let console_input_clone = console_input.clone();
+                set_handler(&signals[0..index], move |received_signals| {
```

Comment:
And you can just put the reference to the Vec in here which will get turned into a slice.

Reply:
Explained the action in previous reply.